kura/examples/org.eclipse.kura.example.ble.tisensortag/src/main/java/org/eclipse/kura/example/ble/tisensortag/BluetoothLe.java

```diff
@@ -298,170 +250,132 @@ void checkScan() {
+                this.enablePres = (Boolean) properties.get(PROPERTY_PRES);
             }
-            if (properties.get(this.PROPERTY_GYRO) != null) {
-                this.enableGyro = (Boolean) properties.get(this.PROPERTY_GYRO);
+            if (properties.get(PROPERTY_GYRO) != null) {
+                this.enableGyro = (Boolean) properties.get(PROPERTY_GYRO);
             }
-            if (properties.get(this.PROPERTY_OPTO) != null) {
-                this.enableOpto = (Boolean) properties.get(this.PROPERTY_OPTO);
+            if (properties.get(PROPERTY_OPTO) != null) {
+                this.enableOpto = (Boolean) properties.get(PROPERTY_OPTO);
             }
-            if (properties.get(this.PROPERTY_BUTTONS) != null) {
-                this.enableButtons = (Boolean) properties.get(this.PROPERTY_BUTTONS);
+            if (properties.get(PROPERTY_BUTTONS) != null) {
+                this.enableButtons = (Boolean) properties.get(PROPERTY_BUTTONS);
             }
-            if (properties.get(this.PROPERTY_REDLED) != null) {
-                this.enableRedLed = (Boolean) properties.get(this.PROPERTY_REDLED);
+            if (properties.get(PROPERTY_REDLED) != null) {
+                this.enableRedLed = (Boolean) properties.get(PROPERTY_REDLED);
             }
-            if (properties.get(this.PROPERTY_GREENLED) != null) {
-                this.enableGreenLed = (Boolean) properties.get(this.PROPERTY_GREENLED);
+            if (properties.get(PROPERTY_GREENLED) != null) {
+                this.enableGreenLed = (Boolean) properties.get(PROPERTY_GREENLED);
             }
-            if (properties.get(this.PROPERTY_BUZZER) != null) {
-                this.enableBuzzer = (Boolean) properties.get(this.PROPERTY_BUZZER);
+            if (properties.get(PROPERTY_BUZZER) != null) {
+                this.enableBuzzer = (Boolean) properties.get(PROPERTY_BUZZER);
             }
-            if (properties.get(this.PROPERTY_TOPIC) != null) {
-                topic = (String) properties.get(this.PROPERTY_TOPIC);
+            if (properties.get(PROPERTY_TOPIC) != null) {
+                this.topic = (String) properties.get(PROPERTY_TOPIC);
             }
-            if (properties.get(this.PROPERTY_INAME) != null) {
-                this.iname = (String) properties.get(this.PROPERTY_INAME);
+            if (properties.get(PROPERTY_INAME) != null) {
+                this.iname = (String) properties.get(PROPERTY_INAME);
             }
         }
     }
 
-    // --------------------------------------------------------------------
-    //
-    // BluetoothLeScanListener APIs
-    //
-    // --------------------------------------------------------------------
-    @Override
-    public void onScanFailed(int errorCode) {
-        logger.error("Error during scan");
-
-    }
-
-    @Override
-    public void onScanResults(List<BluetoothDevice> scanResults) {
-
+    private void doScanCallback(List<BluetoothLeDevice> devices) {
```

Comment:
I removed the callback and I replaced it with the synchronous scan method. Moreover, I tried to split the huge method that reads the SensorTags in multiple little ones.